streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -749,38 +807,12 @@ public void onAssignment(final Assignment assignment, final ConsumerGroupMetadat
             return;
         }
         final int receivedAssignmentMetadataVersion = info.version();
-        final int leaderSupportedVersion = info.latestSupportedVersion();
-
-        if (receivedAssignmentMetadataVersion > usedSubscriptionMetadataVersion) {
-            throw new IllegalStateException(
-                "Sent a version " + usedSubscriptionMetadataVersion
-                    + " subscription but got an assignment with higher version "
-                    + receivedAssignmentMetadataVersion + "."
-            );
-        }
-
-        if (receivedAssignmentMetadataVersion < usedSubscriptionMetadataVersion
-            && receivedAssignmentMetadataVersion >= EARLIEST_PROBEABLE_VERSION) {
+        final int latestCommonlySupportedVersion = info.latestSupportedVersion();
```

Comment:
nit: maybe rename the field in AssignmentInfo as `minSupportedVersion` -- or `commonlySupportedVersion` is also fine -- and also rename the access function names as well as the local fields here? I felt `latestSupportedVersion` is becoming confusing now.

Reply:
Yeah I kept flip-flopping on what I thought a good name would be -- happy to just go with `commonlySupportedVersion`